collector/src/output_frontend.cpp

```diff
@@ -518,6 +569,74 @@ size_t CMonitorOutputFrontend::get_current_sample_measurements() const
     return ntotal_meas;
 }
 
+#ifdef PROMETHEUS_SUPPORT
+void CMonitorOutputFrontend::push_current_sections_to_prometheus()
+{
+    std::string section_name;
+    for (size_t i = 0; i < m_current_sections.size(); i++) {
+        auto& sec = m_current_sections[i];
+        if (sec.m_measurements.empty()) {
+            for (size_t i = 0; i < sec.m_subsections.size(); i++) {
+                auto& subsec = sec.m_subsections[i];
+                section_name = sec.m_name;
+                if (subsec.m_measurements.empty()) {
+                    for (size_t i = 0; i < subsec.m_subsubsections.size(); i++) {
+                        auto& subsubsec = subsec.m_subsubsections[i];
+                        for (size_t n = 0; n < subsubsec.m_measurements.size(); n++) {
+                            auto& measurement = subsubsec.m_measurements[n];
+                            std::string metric_name = section_name + "_" + subsubsec.m_name;
+                            if (subsubsec.m_name != "proc_info")
+                                generate_prometheus_metric(
+                                    metric_name, measurement.m_name.data(), measurement.m_dvalue, subsubsec.m_labels);
+                        }
+                    }
+
+                } else {
+                    for (size_t n = 0; n < subsec.m_measurements.size(); n++) {
+                        auto& measurement = subsec.m_measurements[n];
+                        std::string metric_name;
+                        if ((section_name == "cgroup_network") || (section_name == "cgroup_cpuacct_stats")
```

Comment:
I don't like much hardcoding different rules for specific section names... why cannot we use instead this same rule (metric_name = <section name>, label=metric=<subsection name>) always?